pkgs/build-support/fetchgit/default.nix

```
@@ -11,7 +11,12 @@
   in "${if matched == null then base else builtins.head matched}${appendShort}";
 in
 lib.makeOverridable (lib.fetchers.withNormalizedHash { } (
-{ url, rev ? "HEAD", leaveDotGit ? deepClone
+# NOTE Please document parameter additions or changes in
+#   doc/build-helpers/fetchers.chapter.md
+{ url
+, tag ? null
+, rev ? if tag != null then tag else "HEAD" # FIXME fetching HEAD by default is problematic at best
```

Shouldn't it be using "refs/tags/..." ?

```suggestion
, rev ? if tag != null then "refs/tags/${tag}" else "HEAD" # FIXME fetching HEAD by default is problematic at best
```